Barotrauma/BarotraumaShared/SharedSource/StatusEffects/StatusEffect.cs

```diff
@@ -2281,15 +2285,74 @@ void OnItemSpawned(Item newItem, ItemSpawnInfo itemSpawnInfo)
             }
             if (removeCharacter)
             {
+                void SpawnDespawnContainer(Character character)
+                {
+                    ItemPrefab containerPrefab;
+                    if (saveItemsContainerIdentifier == null) // If no custom identifier is specified for the container item, use the default ones.
+                    {
+                        containerPrefab =
+                        ItemPrefab.Prefabs.Find(me => me.Tags.Contains("despawncontainer")) ??
+                        (MapEntityPrefab.FindByIdentifier("metalcrate".ToIdentifier()) as ItemPrefab);
+                    }
+                    else
+                    {
+                        containerPrefab = ItemPrefab.Prefabs.Find(me => me.Identifier == saveItemsContainerIdentifier);
+                    }
+
+                    if (containerPrefab == null)
+                    {
+                        if (saveItemsContainerIdentifier == null)
+                        {
+                            DebugConsole.NewMessage("Could not spawn a container for a removed character's items. No item with the tag \"despawncontainer\" or the identifier \"metalcrate\" found.", Color.Red);
+                        }
+                        else
+                        {
+                            DebugConsole.NewMessage($"Could not spawn a container for a removed character's items. No item with the identifier “{saveItemsContainerIdentifier}” found.", Color.Red);
+                        }
+                    }
+                    else
+                    {
+                        Entity.Spawner?.AddItemToSpawnQueue(containerPrefab, position, onSpawned: onItemContainerSpawned);
+                    }
+
+                    void onItemContainerSpawned(Item item)
+                    {
+                        if (character.Inventory == null) { return; }
+
+                        item.UpdateTransform();
+                        item.AddTag("name:" + character.Name);
+                        if (character.Info?.Job != null) { item.AddTag($"job:{character.Info.Job.Name}"); }
+
+                        var itemContainer = item?.GetComponent<ItemContainer>();
+                        if (itemContainer == null) { return; }
+                        List<Item> inventoryItems = new List<Item>(character.Inventory.AllItemsMod);
+                        foreach (Item inventoryItem in inventoryItems)
+                        {
+                            if (!itemContainer.Inventory.TryPutItem(inventoryItem, user: null, createNetworkEvent: true))
+                            {
+                                //if the item couldn't be put inside the despawn container, just drop it
+                                inventoryItem.Drop(dropper: character, createNetworkEvent: true);
+                            }
+                        }
+                    }
+                }
```

Comment:
In my opinion this should be moved into a separate method, because this method is already quite massive.